pkg/operator/ceph/object/user/controller.go

```diff
@@ -293,6 +293,25 @@ func (r *ReconcileObjectStoreUser) createOrUpdateCephUser(u *cephv1.CephObjectSt
 	logCreateOrUpdate := fmt.Sprintf("retrieved existing ceph object user %q", u.Name)
 	var user admin.User
 	var err error
+
+	// get the user defined k8s s3 keys if exists
+	secretName := object.GenerateCephUserSecretName(u.Spec.Store, u.Name)
+	namspacedName := types.NamespacedName{Namespace: u.Namespace, Name: secretName}
+	cephObjectStoreUserSecret := &corev1.Secret{}
+	err = r.client.Get(r.clusterInfo.Context, namspacedName, cephObjectStoreUserSecret)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			logger.Debugf("no user secret %q provided for cephobjectuser, using random values", secretName)
+		} else {
+			return errors.Wrapf(err, "failed to get user cephobjectuser secret %q", secretName)
+		}
+	} else {
+		r.userConfig.Keys = []admin.UserKeySpec{
+			{AccessKey: string(cephObjectStoreUserSecret.Data["AccessKey"]),
+				SecretKey: string(cephObjectStoreUserSecret.Data["SecretKey"])},
+		}
+	}
+
 	user, err = r.objContext.AdminOpsClient.GetUser(r.opManagerContext, *r.userConfig)
```

Comment:
This just gets the user. Is there another method instead of `GetUser()` that will create the secret if it doesn't exist yet, or update the secret in case it already existed but is different?

Reply:
Ok great, so this means if it's a new user, the CreateUser will apply the password retrieved from the secret, or if the user already exists, it will update the user with the new secret, right?